water/lib/Dialect/Wave/IR/WaveOps.cpp

```diff
@@ -251,19 +267,42 @@ mlir::LogicalResult wave::IterateOp::verify() {
   }
   for (auto &&[i, iterArg, result] :
        llvm::enumerate(iterArgTypes, resultTypes)) {
-    auto iterArgTensor = llvm::cast<wave::WaveTensorType>(iterArg);
-    auto resultTensor = llvm::cast<wave::WaveTensorType>(result);
-    if (!iterArgTensor.getFullySpecified() || !resultTensor.getFullySpecified())
-      continue;
+    // Handle verification for both wave tensors and vectors
+    auto iterArgTensor = llvm::dyn_cast<wave::WaveTensorType>(iterArg);
+    auto resultTensor = llvm::dyn_cast<wave::WaveTensorType>(result);
+    auto iterArgVector = llvm::dyn_cast<mlir::VectorType>(iterArg);
+    auto resultVector = llvm::dyn_cast<mlir::VectorType>(result);
 
-    auto allDims =
-        llvm::to_vector(llvm::iota_range<int>(0, iterArgTensor.getRank(),
-                                              /*Inclusive=*/false));
     auto istr = std::to_string(i);
-    if (mlir::failed(detail::verifyTypesMatchingDimensions(
-            getLoc(), "iter_args #" + istr, iterArgTensor, allDims,
-            "result #" + istr, resultTensor, allDims)))
-      return mlir::failure();
+
+    // Both are wave tensors - use existing shape verification logic
```

Comment:
"existing" doesn't make sense in the standalone comment when reading the code. It only makes sense when reading the _diff_.